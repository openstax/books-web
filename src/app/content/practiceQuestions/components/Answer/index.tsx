import { HTMLElement } from '@openstax/types/lib.dom';
import React from 'react';
import { FormattedMessage } from 'react-intl';
import { LinkedArchiveTreeSection } from '../../../types';
import { PracticeAnswer } from '../../types';
import { AnswerAlignment, AnswerBlock, AnswerContent, AnswerExcerpt, AnswerLabel, StyledAnswerResult } from './styled';

interface AnswerResultProps {
  showCorrect: boolean;
  isSubmitted: boolean;
  isSelected: boolean;
  isCorrect: boolean;
}

// tslint:disable-next-line: variable-name
const AnswerResult = ({ showCorrect, isSelected, isSubmitted, isCorrect }: AnswerResultProps) => {
  const resultMsgKey = isCorrect
    ? 'i18n:practice-questions:popup:correct'
    : 'i18n:practice-questions:popup:incorrect';

  if (
    !(isSubmitted && isSelected)
    && !(isSubmitted && showCorrect && isCorrect)
  ) {
    return null;
  }

  return <StyledAnswerResult isCorrect={isCorrect}>
    <FormattedMessage id={resultMsgKey}>
      {(msg: string) => msg}
    </FormattedMessage>
  </StyledAnswerResult>;
};

interface AnswerProps {
  isSelected: boolean;
  isSubmitted: boolean;
  showCorrect: boolean;
  answer: PracticeAnswer;
  choiceIndicator: string;
  onSelect: () => void;
  source: LinkedArchiveTreeSection;
}

// tslint:disable-next-line: variable-name
const Answer = ({
  answer,
  showCorrect,
  choiceIndicator,
  isSubmitted,
  isSelected,
  onSelect,
}: AnswerProps) => {
  const answerRef = React.useRef<HTMLElement>(null);
  const isCorrect = answer.correctness === '1.0';

  React.useEffect(() => {
    if (showCorrect && isCorrect && answerRef.current) {
      answerRef.current.focus();
    }
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [showCorrect, isCorrect]);

  return <AnswerBlock
    showCorrect={showCorrect}
    isCorrect={isCorrect}
    isSubmitted={isSubmitted}
    isSelected={isSelected}
    onClick={onSelect}
  >
    <FormattedMessage
      id='i18n:practice-questions:popup:answers:choice'
      values={{choiceIndicator: choiceIndicator.toUpperCase()}}
    >{(msg: string) =>
      <React.Fragment>
        <input
          id={choiceIndicator}
          type='radio'
          name={choiceIndicator}
          checked={isSelected}
          disabled={isSubmitted}
          onChange={onSelect}
        />
        <AnswerLabel ref={answerRef} aria-label={msg} for={choiceIndicator} tabIndex={-1}>
          {choiceIndicator}
        </AnswerLabel>
      </React.Fragment>
    }</FormattedMessage>
    <AnswerAlignment>
      <AnswerContent tabIndex={0}>
        <AnswerExcerpt>{answer.content_html}</AnswerExcerpt>
        <AnswerResult
          showCorrect={showCorrect}
          isSelected={isSelected}
          isSubmitted={isSubmitted}
          isCorrect={isCorrect}
        />
      </AnswerContent>
    </AnswerAlignment>
  </AnswerBlock>;
};

export default Answer;
