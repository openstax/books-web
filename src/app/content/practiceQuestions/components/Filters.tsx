import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import ChapterFilter from '../../components/popUp/ChapterFilter';
import Filters, { FilterDropdown, FiltersTopBar } from '../../components/popUp/Filters';
import * as contentSelectors from '../../selectors';
import { LinkedArchiveTreeSection } from '../../types';
import { findArchiveTreeNodeById } from '../../utils/archiveTreeUtils';
import { setSelectedSection } from '../actions';
import * as selectors from '../selectors';

export default () => {
  const [open, setOpen] = React.useState(false);
  const locationFilters = useSelector(selectors.practiceQuestionsLocationFilters);
  const selectedSection = useSelector(selectors.selectedSection);
  const book = useSelector(contentSelectors.book);
  const [isOpenChapterId, setIsOpenChapterId] = React.useState(selectedSection ? selectedSection.parent.id : null);
  const dispatch = useDispatch();
  const setFilters = React.useCallback(({ locationIds }: { locationIds: string[] }) => {
    const section = book && locationIds.length
      ? findArchiveTreeNodeById(book.tree, locationIds[0]) as LinkedArchiveTreeSection
      : null;
    dispatch(setSelectedSection(section));
    setOpen(false);
  }, [book, dispatch]);
  const selectedLocationFilters = React.useMemo(
    () => new Set(selectedSection ? [selectedSection.id] : []),
    [selectedSection]);

  return <Filters>
    <FiltersTopBar>
      <FilterDropdown
        label='i18n:practice-questions:popup:filters:chapters'
        ariaLabelId='i18n:practice-questions:popup:filters:filter-by:aria-label'
        open={open}
        setOpen={setOpen}
      >
        <ChapterFilter
          locationFilters={locationFilters}
          selectedLocationFilters={selectedLocationFilters}
          setFilters={setFilters}
          hideAllOrNone={true}
          isOpenChapterId={isOpenChapterId}
          onChapterToggleClick={(id: string) => setIsOpenChapterId((prev) => prev === id ? null : id)}
        />
      </FilterDropdown>
    </FiltersTopBar>
  </Filters>;
};
