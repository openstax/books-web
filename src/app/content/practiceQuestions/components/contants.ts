import theme from '../../../theme';
import { PracticeQuestionStyles } from '../types';

const correctAnswerStyle: PracticeQuestionStyles = {
  background: '#77AF42',
  border: '#77AF42',
  borderHovered: '#77AF42',
  fontColor: theme.color.neutral.base,
  fontColorActive: '#77AF42',
  label: 'correct',
};

const incorrectAnswerStyle: PracticeQuestionStyles = {
  background: theme.color.primary.red.base,
  border: theme.color.primary.red.base,
  borderHovered: theme.color.primary.red.base,
  fontColor: theme.color.neutral.base,
  fontColorActive: theme.color.primary.red.base,
  label: 'incorrect',
};

const selectedAnswerStyle: PracticeQuestionStyles = {
  background: theme.color.secondary.lightBlue.base,
  border: theme.color.secondary.lightBlue.base,
  borderHovered: theme.color.secondary.lightBlue.base,
  fontColor: theme.color.neutral.base,
  fontColorActive: theme.color.secondary.lightBlue.base,
  label: 'selected',
};

const unselectedAnswerStyle: PracticeQuestionStyles = {
  background: theme.color.neutral.base,
  border: '#C6C6C6',
  borderHovered: theme.color.secondary.lightBlue.base,
  fontColor: '#606163',
  fontColorActive: '#C6C6C6',
  label: 'unselected',
};

export default {
  correct: correctAnswerStyle,
  incorrect: incorrectAnswerStyle,
  selected: selectedAnswerStyle,
  unselected: unselectedAnswerStyle,
};
