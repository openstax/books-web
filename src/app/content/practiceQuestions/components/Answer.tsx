import React from 'react';
import { FormattedMessage } from 'react-intl';
import styled, { css } from 'styled-components/macro';
import { textRegularStyle } from '../../../components/Typography';
import theme from '../../../theme';
import ContentExcerpt from '../../components/ContentExcerpt';
import { LinkedArchiveTreeSection } from '../../types';
import { PracticeAnswer, PracticeQuestionStyles } from '../types';
import contants from './contants';

interface AnswerProps {
  isSelected: boolean;
  isSubmitted: boolean;
  showCorrect: boolean;
  answer: PracticeAnswer;
  choiceIndicator: string;
  onSelect: () => void;
  source: LinkedArchiveTreeSection;
}

// tslint:disable-next-line: variable-name
const AnswerExcerpt = styled.span`
  ${textRegularStyle}
  width: 100%;
  padding: 0;
`;

// tslint:disable-next-line: variable-name
const AnswerAlignment = styled.div`
  display: flex;
  align-items: center;
  min-height: 4rem;
`;

// tslint:disable-next-line: variable-name
const AnswerContent = styled.div`
  margin-left: 1.6rem;
`;

// tslint:disable-next-line: variable-name
const AnswerLabel = styled.label`
  background-color: ${(props: {style: PracticeQuestionStyles}) => props.style.focused};
  border: 1.5px solid ${(props: {style: PracticeQuestionStyles}) => props.style.passive};
  min-width: 4rem;
  min-height: 4rem;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  color: ${(props: {style: PracticeQuestionStyles}) => props.style.fontColor};

  input {
    position: absolute;
    opacity: 0;
    height: 0;
    width: 0;
  }
`;

// tslint:disable-next-line: variable-name
export const AnswerBlock = styled.div`
  padding: 0 2rem 2.4rem;
  display: flex;
  align-items: flex-start;
  cursor: pointer;

  &:hover {
    ${AnswerLabel} {
      border-color: ${(props: {style: PracticeQuestionStyles}) => props.style.hovered};
    }
  }

  ${theme.breakpoints.mobile(css`
    padding: 0 1rem 2.4rem;

    ${ContentExcerpt} {
      padding: 0;
    }
  `)}
`;

// tslint:disable-next-line: variable-name
const Answer = (props: AnswerProps) => {
  const answerStyle = props.isSelected ? contants.selectedAnswerStyle : contants.unselectedAnswerStyle;

  return <AnswerBlock style={answerStyle} onClick={props.onSelect}>
    <FormattedMessage
      id='i18n:practice-questions:popup:answers:choice'
      values={{choiceIndicator: props.choiceIndicator.toUpperCase()}}
    >{(msg: string) =>
      <AnswerLabel style={answerStyle} htmlFor={props.answer.id} aria-label={msg}>
        {props.choiceIndicator}
        <input
          type='radio'
          id={`${props.answer.id}`}
          name={props.choiceIndicator}
          checked={props.isSelected}
          onChange={props.onSelect}
        />
      </AnswerLabel>
    }</FormattedMessage>
    <AnswerAlignment>
      <AnswerContent>
        <AnswerExcerpt>{props.answer.content_html}</AnswerExcerpt>
      </AnswerContent>
    </AnswerAlignment>
  </AnswerBlock>;
};

export default Answer;
