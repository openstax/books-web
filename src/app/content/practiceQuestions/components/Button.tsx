import styled from 'styled-components/macro';
import { PlainButton } from '../../../components/Button';
import theme from '../../../theme';

interface ButtonProps {
  withoutBackground: boolean;
}

export default styled(PlainButton)`
  backgrouond-color: ${theme.color.primary.orange.base};
  color: ${theme.color.text.white};
  font-size: 1.6rem;
  font-weight: bold;
  height: 5rem;
  padding: 0 3rem;
  ${(props: ButtonProps) => {
    if (props.withoutBackground) {
      return `
        backgrouond-color: transparent;
        color: #027eb5;
      `;
    }
  }}
`;
