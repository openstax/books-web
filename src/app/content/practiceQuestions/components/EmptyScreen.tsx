import React from 'react';
import { FormattedMessage } from 'react-intl';
import { useDispatch } from 'react-redux';
import styled from 'styled-components/macro';
import { textRegularStyle } from '../../../components/Typography';
import { LinkedArchiveTreeSection } from '../../types';
import { setSelectedSection } from '../actions';
import NextSectionMessage from './NextSectionMessage';

// tslint:disable-next-line: variable-name
const StyledEmptyScreen = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  max-width: 38rem;
  margin: 2rem auto;
  flex: 1;
  text-align: center;
  ${textRegularStyle}

  > span {
    margin-bottom: 3rem;
  }
`;

interface EmptyScreenProps {
  nextSection: LinkedArchiveTreeSection;
}

// tslint:disable-next-line: variable-name
const EmptyScreen = ({ nextSection }: EmptyScreenProps) => {
  const dispatch = useDispatch();

  return <StyledEmptyScreen>
    <span>
      <FormattedMessage id='i18n:practice-questions:popup:empty:message'>
        {(msg: string) => msg}
      </FormattedMessage>
    </span>
    <NextSectionMessage
      nextSection={nextSection}
      messageKey='i18n:practice-questions:popup:empty:next-section'
      onClick={() => dispatch(setSelectedSection(nextSection))}
    />
  </StyledEmptyScreen>;
};

export default EmptyScreen;
