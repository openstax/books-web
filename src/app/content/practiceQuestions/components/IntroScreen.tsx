import React from 'react';
import { FormattedMessage } from 'react-intl';
import { useDispatch, useSelector } from 'react-redux';
import styled from 'styled-components/macro';
import { textRegularSize } from '../../../components/Typography';
import { nextQuestion } from '../actions';
import * as pqSelectors from '../selectors';
import PQButton from './PQButton';

// tslint:disable-next-line: variable-name
const IntroScreenWrapper = styled.div`
  text-align: center;

  ${PQButton} {
    display: block;
    margin: 2rem auto;
  }
`;

// tslint:disable-next-line: variable-name
const IntroScreenMessage = styled.span`
  ${textRegularSize}
  margin: 4rem 0;
`;

// tslint:disable-next-line: variable-name
const IntroScreen = () => {
  const questionsCount = useSelector(pqSelectors.questionsCount);
  const dispatch = useDispatch();

  return <IntroScreenWrapper>
    <IntroScreenMessage>
      <FormattedMessage id='i18n:practice-questions:popup:intro:message' values={{ questions: questionsCount }}>
        {(msg: string) => msg}
      </FormattedMessage>
    </IntroScreenMessage>
    <PQButton
      variant='primary'
      size='large'
      onClick={() => dispatch(nextQuestion())}
      data-analytics-label='Start now'
    >
      <FormattedMessage id='i18n:practice-questions:popup:intro:start'>
        {(msg: string) => msg}
      </FormattedMessage>
    </PQButton>
  </IntroScreenWrapper>;
};

export default IntroScreen;
