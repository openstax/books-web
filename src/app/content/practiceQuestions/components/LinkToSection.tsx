import React from 'react';
import { FormattedMessage } from 'react-intl';
import { useSelector } from 'react-redux';
import styled, { css } from 'styled-components/macro';
import { linkColor } from '../../../components/Typography';
import theme from '../../../theme';
import * as contentSelectors from '../../selectors';
import { LinkedArchiveTreeSection } from '../../types';
import { getBookPageUrlAndParams } from '../../utils/urlUtils';

// tslint:disable-next-line: variable-name
export const StyledContentLink = styled.a`
  display: block;
  font-size: 1.4rem;
  color: #929292;
  margin-top: 2.5rem;
  text-decoration: none;

  > span {
    color: ${linkColor};

    &::before {
      content: " ";
    }
  }

  ${theme.breakpoints.mobile(css`
    margin: 1.2rem;
  `)}
`;

interface LinkToSectionProps {
  section: LinkedArchiveTreeSection | null;
}

// tslint:disable-next-line: variable-name
const LinkToSection = ({ section }: LinkToSectionProps) => {
  const book = useSelector(contentSelectors.book);

  const linkToTheSection = React.useMemo(() => {
    return book && section ? getBookPageUrlAndParams(book, section).url : null;
  }, [book, section]);

  if (!section || !linkToTheSection) { return null; }

  return <StyledContentLink href={linkToTheSection} target='_blank' data-analytics-label='Go to link' >
    <FormattedMessage id='i18n:practice-questions:popup:read'>
      {(msg: string) => msg}
    </FormattedMessage>
    <span dangerouslySetInnerHTML={{ __html: section.title }} />
  </StyledContentLink>;
};

export default LinkToSection;
