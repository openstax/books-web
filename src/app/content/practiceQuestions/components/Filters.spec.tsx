import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import createTestServices from '../../../../test/createTestServices';
import createTestStore from '../../../../test/createTestStore';
import { DropdownToggle } from '../../../components/Dropdown';
import * as Services from '../../../context/Services';
import MessageProvider from '../../../MessageProvider';
import { Store } from '../../../types';
import { StyledDetails, StyledSectionItem, StyledSummary } from '../../components/popUp/ChapterFilter';
import { LinkedArchiveTreeSection } from '../../types';
import { setSelectedSection } from '../actions';
import * as selectors from '../selectors';
import { PracticeQuestionsLocationFilters } from '../types';
import Filters from './Filters';

describe('Filters', () => {
  let store: Store;
  let services: ReturnType<typeof createTestServices>;
  let dispatch: jest.SpyInstance;
  let render: () => JSX.Element;
  const mockLocationFilters = new Map([
    ['chapterId', {
      chapter: { id: 'chapterId', title: 'chapterId' },
      sections: [{ id: 'section1', title: 'section1' }, { id: 'section2', title: 'section2' }],
    }],
    ['chapterId2', {
      chapter: { id: 'chapterId2', title: 'chapterId2' },
      sections: [{ id: 'section21', title: 'section21' }, { id: 'section22', title: 'section22' }],
    }],
  ]) as PracticeQuestionsLocationFilters;

  beforeEach(() => {
    store = createTestStore();
    services = createTestServices();
    dispatch = jest.spyOn(store, 'dispatch');
    render = () => <Provider store={store}>
      <Services.Provider value={services}>
        <MessageProvider>
          <Filters />
        </MessageProvider>
      </Services.Provider>
    </Provider>;
  });

  it('ChapterFilterWithToggling matches snapshot when it is closed', () => {
    const mockFilters = jest.spyOn(selectors, 'practiceQuestionsLocationFilters')
      .mockReturnValue(mockLocationFilters);
    const mockSection = jest.spyOn(selectors, 'selectedSection')
      .mockReturnValue({ id: 'section21', parent: { id: 'chapterId2' } } as LinkedArchiveTreeSection);

    const component = renderer.create(render());

    expect(component).toMatchSnapshot();

    mockFilters.mockReset();
    mockSection.mockReset();
  });

  it('ChapterFilterWithToggling matches snapshot when open without selected section', () => {
    const mockFilters = jest.spyOn(selectors, 'practiceQuestionsLocationFilters')
      .mockReturnValue(mockLocationFilters);

    const component = renderer.create(render());

    renderer.act(() => {
      const chapterFilterToggle = component.root.findByType(DropdownToggle);
      chapterFilterToggle.props.onClick();
    });

    expect(component).toMatchSnapshot();

    mockFilters.mockReset();
  });

  it('ChapterFilterWithToggling matches snapshot when open with selected section', () => {
    const mockFilters = jest.spyOn(selectors, 'practiceQuestionsLocationFilters')
      .mockReturnValue(mockLocationFilters);
    const mockSection = jest.spyOn(selectors, 'selectedSection')
      .mockReturnValue({ id: 'section21', parent: { id: 'chapterId2' } } as LinkedArchiveTreeSection);

    const component = renderer.create(render());

    renderer.act(() => {
      const chapterFilterToggle = component.root.findByType(DropdownToggle);
      chapterFilterToggle.props.onClick();
    });

    expect(component).toMatchSnapshot();

    mockFilters.mockReset();
    mockSection.mockReset();
  });

  it('ChapterFilter with chapters and sections works properly', () => {
    const mockFilters = jest.spyOn(selectors, 'practiceQuestionsLocationFilters')
      .mockReturnValue(mockLocationFilters);
    const mockSection = jest.spyOn(selectors, 'selectedSection')
      .mockReturnValue({ id: 'section21', parent: { id: 'chapterId2' } } as LinkedArchiveTreeSection);

    const component = renderer.create(render());

    renderer.act(() => {
      const chapterFilterToggle = component.root.findByType(DropdownToggle);
      chapterFilterToggle.props.onClick();
    });

    const [details1, details2] = component.root.findAllByType(StyledDetails);
    expect(details1.props.open).toEqual(false);
    expect(details2.props.open).toEqual(true);

    const mockPreventDefault = jest.fn();

    renderer.act(() => {
      const summary = details2.findByType(StyledSummary);
      summary.props.onClick({ preventDefault: mockPreventDefault });
    });

    expect(details1.props.open).toEqual(false);
    expect(details2.props.open).toEqual(false);

    renderer.act(() => {
      const summary = details1.findByType(StyledSummary);
      summary.props.onClick({ preventDefault: mockPreventDefault });
    });

    expect(mockPreventDefault).toHaveBeenCalled();
    expect(details1.props.open).toEqual(true);
    expect(details2.props.open).toEqual(false);

    renderer.act(() => {
      const [section1] = details1.findAllByType(StyledSectionItem);
      section1.props.onClick();
    });

    expect(dispatch).toHaveBeenCalledWith(setSelectedSection({ id: 'section1', title: 'section1' } as any));

    mockFilters.mockReset();
    mockSection.mockReset();
  });
});
