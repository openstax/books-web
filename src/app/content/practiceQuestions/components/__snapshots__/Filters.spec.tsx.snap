// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters ChapterFilter matches snapshot when it is closed 1`] = `
<div
  className="Filters-sc-1fn81fx-3 gZlQQL"
>
  <div
    className="Filters__FiltersTopBar-sc-1fn81fx-2 fWMshY"
  >
    <div
      className="Dropdown-rmc6yw-5 fBPFfC Dropdown__TabHiddenDropDown-rmc6yw-1 kbMrSv"
    >
      <button
        aria-label="Filter PQ by Chapter & Section"
        className="Dropdown__DropdownToggle-rmc6yw-0 gpiKKF Filters__Toggle-sc-1fn81fx-1 liapzj Button__PlainButton-ayg7nk-2 chgswm"
        onClick={[Function]}
      >
        <div
          tabIndex={-1}
        >
          Chapter & Section
          <svg
            aria-hidden="true"
            className="Filters__AngleIcon-sc-1fn81fx-0 jYDriK StyledIconBase-sc-bdy9j4 cCvWXa"
            direction="down"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 320 512"
          >
            <path
              d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              fill="currentColor"
            />
          </svg>
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`Filters ChapterFilter matches snapshot when open with selected section 1`] = `
<div
  className="Filters-sc-1fn81fx-3 gZlQQL"
>
  <div
    className="Filters__FiltersTopBar-sc-1fn81fx-2 fWMshY"
  >
    <div
      className="Dropdown-rmc6yw-5 fBPFfC Dropdown__TabHiddenDropDown-rmc6yw-1 kbMrSv"
    >
      <button
        aria-label="Filter PQ by Chapter & Section"
        className="Dropdown__DropdownToggle-rmc6yw-0 gpiKKF Filters__Toggle-sc-1fn81fx-1 gsKueT Button__PlainButton-ayg7nk-2 chgswm"
        onClick={[Function]}
      >
        <div
          tabIndex={-1}
        >
          Chapter & Section
          <svg
            aria-hidden="true"
            className="Filters__AngleIcon-sc-1fn81fx-0 jKNYyQ StyledIconBase-sc-bdy9j4 cCvWXa"
            direction="up"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 320 512"
          >
            <path
              d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              fill="currentColor"
            />
          </svg>
        </div>
      </button>
      <div
        className="ChapterFilter-tcdlu-6 Filters__StyledChapterFilters-sc-1lvdmlz-0 jvrNgK"
        tabIndex={-1}
      >
        <div
          className="ChapterFilter__Row-tcdlu-0 nqIYu"
        >
          <div
            className="ChapterFilter__Column-tcdlu-1 dKPjiE"
          >
            <details
              className="ChapterFilter__StyledDetails-tcdlu-3 kzZRsL"
              open={false}
            >
              <summary
                className="ChapterFilter__StyledSummary-tcdlu-4 ieEWMq"
                onClick={[Function]}
              >
                <span
                  className="ChapterFilter__ChapterTitle-tcdlu-2 fmaoPu"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "chapterId",
                    }
                  }
                />
                <svg
                  aria-hidden="true"
                  className="Filters__AngleIcon-sc-1fn81fx-0 jYDriK StyledIconBase-sc-bdy9j4 cCvWXa"
                  direction="down"
                  fill="currentColor"
                  focusable="false"
                  viewBox="0 0 320 512"
                >
                  <path
                    d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                    fill="currentColor"
                  />
                </svg>
              </summary>
              <div>
                <button
                  className="Button__PlainButton-ayg7nk-2 ChapterFilter__StyledSectionItem-tcdlu-5 fQHjjn"
                  onClick={[Function]}
                >
                  <span
                    className="ChapterFilter__ChapterTitle-tcdlu-2 fmaoPu"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "section1",
                      }
                    }
                  />
                </button>
                <button
                  className="Button__PlainButton-ayg7nk-2 ChapterFilter__StyledSectionItem-tcdlu-5 fQHjjn"
                  onClick={[Function]}
                >
                  <span
                    className="ChapterFilter__ChapterTitle-tcdlu-2 fmaoPu"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "section2",
                      }
                    }
                  />
                </button>
              </div>
            </details>
            <details
              className="ChapterFilter__StyledDetails-tcdlu-3 kzZRsL"
              open={true}
            >
              <summary
                className="ChapterFilter__StyledSummary-tcdlu-4 ieEWMq"
                onClick={[Function]}
              >
                <span
                  className="ChapterFilter__ChapterTitle-tcdlu-2 fmaoPu"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "chapterId2",
                    }
                  }
                />
                <svg
                  aria-hidden="true"
                  className="Filters__AngleIcon-sc-1fn81fx-0 jKNYyQ StyledIconBase-sc-bdy9j4 cCvWXa"
                  direction="up"
                  fill="currentColor"
                  focusable="false"
                  viewBox="0 0 320 512"
                >
                  <path
                    d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                    fill="currentColor"
                  />
                </svg>
              </summary>
              <div>
                <button
                  className="Button__PlainButton-ayg7nk-2 ChapterFilter__StyledSectionItem-tcdlu-5 lhdDtN"
                  onClick={[Function]}
                >
                  <span
                    className="ChapterFilter__ChapterTitle-tcdlu-2 fmaoPu"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "section21",
                      }
                    }
                  />
                </button>
              </div>
            </details>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Filters ChapterFilter matches snapshot when open without selected section 1`] = `
<div
  className="Filters-sc-1fn81fx-3 gZlQQL"
>
  <div
    className="Filters__FiltersTopBar-sc-1fn81fx-2 fWMshY"
  >
    <div
      className="Dropdown-rmc6yw-5 fBPFfC Dropdown__TabHiddenDropDown-rmc6yw-1 kbMrSv"
    >
      <button
        aria-label="Filter PQ by Chapter & Section"
        className="Dropdown__DropdownToggle-rmc6yw-0 gpiKKF Filters__Toggle-sc-1fn81fx-1 gsKueT Button__PlainButton-ayg7nk-2 chgswm"
        onClick={[Function]}
      >
        <div
          tabIndex={-1}
        >
          Chapter & Section
          <svg
            aria-hidden="true"
            className="Filters__AngleIcon-sc-1fn81fx-0 jKNYyQ StyledIconBase-sc-bdy9j4 cCvWXa"
            direction="up"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 320 512"
          >
            <path
              d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              fill="currentColor"
            />
          </svg>
        </div>
      </button>
      <div
        className="ChapterFilter-tcdlu-6 Filters__StyledChapterFilters-sc-1lvdmlz-0 jvrNgK"
        tabIndex={-1}
      >
        <div
          className="ChapterFilter__Row-tcdlu-0 nqIYu"
        >
          <div
            className="ChapterFilter__Column-tcdlu-1 dKPjiE"
          >
            <details
              className="ChapterFilter__StyledDetails-tcdlu-3 kzZRsL"
              open={false}
            >
              <summary
                className="ChapterFilter__StyledSummary-tcdlu-4 ieEWMq"
                onClick={[Function]}
              >
                <span
                  className="ChapterFilter__ChapterTitle-tcdlu-2 fmaoPu"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "chapterId",
                    }
                  }
                />
                <svg
                  aria-hidden="true"
                  className="Filters__AngleIcon-sc-1fn81fx-0 jYDriK StyledIconBase-sc-bdy9j4 cCvWXa"
                  direction="down"
                  fill="currentColor"
                  focusable="false"
                  viewBox="0 0 320 512"
                >
                  <path
                    d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                    fill="currentColor"
                  />
                </svg>
              </summary>
              <div>
                <button
                  className="Button__PlainButton-ayg7nk-2 ChapterFilter__StyledSectionItem-tcdlu-5 fQHjjn"
                  onClick={[Function]}
                >
                  <span
                    className="ChapterFilter__ChapterTitle-tcdlu-2 fmaoPu"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "section1",
                      }
                    }
                  />
                </button>
                <button
                  className="Button__PlainButton-ayg7nk-2 ChapterFilter__StyledSectionItem-tcdlu-5 fQHjjn"
                  onClick={[Function]}
                >
                  <span
                    className="ChapterFilter__ChapterTitle-tcdlu-2 fmaoPu"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "section2",
                      }
                    }
                  />
                </button>
              </div>
            </details>
            <details
              className="ChapterFilter__StyledDetails-tcdlu-3 kzZRsL"
              open={false}
            >
              <summary
                className="ChapterFilter__StyledSummary-tcdlu-4 ieEWMq"
                onClick={[Function]}
              >
                <span
                  className="ChapterFilter__ChapterTitle-tcdlu-2 fmaoPu"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "chapterId2",
                    }
                  }
                />
                <svg
                  aria-hidden="true"
                  className="Filters__AngleIcon-sc-1fn81fx-0 jYDriK StyledIconBase-sc-bdy9j4 cCvWXa"
                  direction="down"
                  fill="currentColor"
                  focusable="false"
                  viewBox="0 0 320 512"
                >
                  <path
                    d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                    fill="currentColor"
                  />
                </svg>
              </summary>
              <div>
                <button
                  className="Button__PlainButton-ayg7nk-2 ChapterFilter__StyledSectionItem-tcdlu-5 fQHjjn"
                  onClick={[Function]}
                >
                  <span
                    className="ChapterFilter__ChapterTitle-tcdlu-2 fmaoPu"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "section21",
                      }
                    }
                  />
                </button>
              </div>
            </details>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
